test/utils.go

```diff
@@ -20,3 +26,131 @@ func SkipIfFalse(c *check.C, conditions ...VerifyCondition) {
+
+	CheckRespStatus(c, resp, 204)
+}
+
+// StartContainer starts the container.
+func StartContainer(c *check.C, cname string) (*http.Response, error) {
+	return request.Post("/containers/" + cname + "/start")
+}
+
+// DelContainerForceOk forcely deletes the container and asserts success.
+func DelContainerForceOk(c *check.C, cname string) {
+	resp, err := DelContainerForce(c, cname)
+	c.Assert(err, check.IsNil)
+
+	CheckRespStatus(c, resp, 204)
+}
+
+// DelContainer forcely deletes the container.
+func DelContainerForce(c *check.C, cname string) (*http.Response, error) {
+	q := url.Values{}
+	q.Add("force", "true")
+	return request.Delete("/containers/"+cname, request.WithQuery(q))
+}
+
+// StopContainerOk stops the container and asserts success..
+func StopContainerOk(c *check.C, cname string) {
+	resp, err := StopContainer(c, cname)
+	c.Assert(err, check.IsNil)
+
+	CheckRespStatus(c, resp, 204)
+}
+
+// StopContainer stops the container.
+func StopContainer(c *check.C, cname string) (*http.Response, error) {
+	return request.Post("/containers/" + cname + "/stop")
+}
+
+// PuseContainerOk pauses the container and asserts success..
+func PauseContainerOk(c *check.C, cname string) {
+	resp, err := PauseContainer(c, cname)
+	c.Assert(err, check.IsNil)
+
+	CheckRespStatus(c, resp, 204)
+}
+
+// PauseContainer pauses the container.
+func PauseContainer(c *check.C, cname string) (*http.Response, error) {
+	return request.Post("/containers/" + cname + "/pause")
+}
+
+// UnpuseContainerOk unpauses the container and asserts success..
+func UnpauseContainerOk(c *check.C, cname string) {
+	resp, err := UnpauseContainer(c, cname)
+	c.Assert(err, check.IsNil)
+
+	CheckRespStatus(c, resp, 204)
+}
+
+// UnpauseContainer unpauses the container.
+func UnpauseContainer(c *check.C, cname string) (*http.Response, error) {
+	return request.Post("/containers/" + cname + "/unpause")
+}
+
+// CheckRespStatus checks the http.Response.Status is equal to status.
+func CheckRespStatus(c *check.C, resp *http.Response, status int) {
+	defer resp.Body.Close()
+	got := types.Error{}
+	request.DecodeBody(&got, resp.Body)
+	c.Assert(resp.Status, check.Equals, status, check.Commentf("Error: %s", got.Message))
+}
+
+// IsContainerCreated returns true is container's state is created.
+func IsContainerCreated(c *check.C, cname string) bool {
+	return isContainerStateEqual(c, cname, "created")
+}
+
+// IsContainerRunning returns true is container's state is running.
+func IsContainerRunning(c *check.C, cname string) bool {
+	return isContainerStateEqual(c, cname, "running")
+}
+
+func isContainerStateEqual(c *check.C, cname string, status string) bool {
+	resp, err := request.Get("/containers/" + cname + "/json")
+	c.Assert(err, check.IsNil)
+	c.Assert(resp.StatusCode, check.Equals, 200)
+
+	defer resp.Body.Close()
+
+	got := types.ContainerJSON{}
+	err = request.DecodeBody(&got, resp.Body)
+	c.Assert(err, check.IsNil)
+
+	return string(got.State.Status) == status
```

Comment:
It may panic, since `State *ContainerState`. We had better detect State is non-nil first.